packages/next/client/head-manager.ts

```diff
@@ -28,14 +34,34 @@ function reactElementToDOM({ type, props }: JSX.Element): HTMLElement {
 
   const { children, dangerouslySetInnerHTML } = props
   if (dangerouslySetInnerHTML) {
-    el.innerHTML = dangerouslySetInnerHTML.__html || ''
+    if (type === 'script') {
```

Comment:
There may be users relying on this behaviour, so this is a breaking change (regardless of whether apps enforce Trusted Types or not). I recommend splitting this out into a separate PR so that we/the Next.js team can discuss this issue in particular and how best to land the change.

Reply:
Fixed